hyrisecockpit/database_manager/job/get_loaded_tables.py

```diff
@@ -0,0 +1,24 @@
+"""This job return loaded tables for scale factor."""
+from typing import List
+
+from psycopg2 import DatabaseError, InterfaceError
+
+
+def get_loaded_tables_for_scale_factor(
+    tables: List[str], benchmark: str, scale_factor: str, connection_factory
+) -> List[str]:
+    """Check which tables exists and which not."""
+    loaded_tables: List = []
+    try:
+        with connection_factory.create_cursor() as cur:
+            cur.execute("show tables;", None)
+            results = cur.fetchall()
+            all_loaded_tables = [row[0] for row in results] if results else []
+            for table in tables:
+                if f"{table}_{benchmark}_{scale_factor}" in all_loaded_tables:
+                    loaded_tables.append(table)
+
+    except (DatabaseError, InterfaceError):
+        return []
```

Comment:
How about returning `None` instead of an empty list if an error occurs? An empty list would mean that the query returned no tuples (thus no tables are loaded).

```suggestion
) -> Optional[List[str]]:
    """Check which tables exists and which not."""
    loaded_tables: List = []
    try:
        with connection_factory.create_cursor() as cur:
            cur.execute("show tables;", None)
            results = cur.fetchall()
            all_loaded_tables = [row[0] for row in results] if results else None
            for table in tables:
                if f"{table}_{benchmark}_{scale_factor}" in all_loaded_tables:
                    loaded_tables.append(table)

    except (DatabaseError, InterfaceError):
        return None
```

Reply:
I agree that a return of None would be cleaner. But if we do that every component the uses the `get_loaded_tables_for_scale_factor ` function needs to handle the case that we can't access the hyrise because of its malfunctioning (this would be of cause totally our fault because it is theoretically impossible that a horse would crash 👀 ). For the moment we are using the low levelest hyrise error handling as possible (so handle it as soon as possible). So we are just saying that if the hyrise is crashed that there are no tables.

Reply:
I see your point Alex. However, I rather agree with Fabi as we also deal with this in other parts of the code (for example when we request the plugin settings) and I think that the effort is worth it.

Reply:
Ok, but I would suggest doing it in a separate PR. Is that ok with you?